File: hackerrank.com/problem-solving/birthday_cake_candles.js
// https://www.hackerrank.com/challenges/birthday-cake-candles/problem

function birthdayCakeCandles(ar) {
  ar = ar.sort((a, b) => b - a)
  a = 0
  while (ar[++a] === ar[0] && a < ar.length) { }
  return a
}

function birthdayCakeCandles2(ar) {
  var max = Math.max(...ar)
  return ar.filter(x => x === max).length
}


function main(inputData) {
  const lines = inputData.split("\n");
  const n = parseInt(lines[0])
  const ar = lines[1].split(' ').map(x => parseInt(x))
  console.log(birthdayCakeCandles2(ar))
}

process.stdin.resume();
process.stdin.setEncoding("ascii");
let _input = '';
process.stdin.on("data", input => {
  _input += input;
});

process.stdin.on("end", () => {
 main(_input);
});

File: hackerrank.com/problem-solving/birthday_cake_candles.scala
// # https://www.hackerrank.com/challenges/birthday-cake-candles/problem


import scala.io.StdIn._

object Solution {

  def birthdayCakeCandles(ar: Array[Int]): Int = {
    val tallest = ar.max
    ar.count(x => x == tallest)
  }

  def birthdayCakeCandles2(ar: Array[Int]): Int = {
    ar.groupBy(identity).map(t => (t._1, t._2.length)).maxBy(_._1)._2
  }

  def birthdayCakeCandles3(ar: Array[Int]): Int = {
    ar.groupBy(identity).maxBy(_._1)._2.length
  }

  def birthdayCakeCandles4(ar: Array[Int]): Int = {
    val tallest = ar.reduceLeft(_ max _)
    ar.filter(_ == tallest).length
  }

  def main(args: Array[String]) {
    val n = readInt
    val ar = readLine.split(" ").map(_.trim.toInt)
    println(birthdayCakeCandles3(ar))
    
  }
}